src/main.rs

```diff
@@ -0,0 +1,228 @@
+	let qw = cos(0.5 * angle);
+	let qx = rx / axis_norm * sin_angle;
+	let qy = ry / axis_norm * sin_angle;
+	let qz = rz / axis_norm * sin_angle;
+
+	let euler_x = atan2(2.0 * (qw * qx - qy * qz), 1.0 - 2.0 * (qx * qx + qy * qy));
+	let euler_y = asin(2.0 * (qx * qz + qw * qy));
+	let euler_z = atan2(2.0 * (qw * qz - qx * qy), 1.0 - 2.0 * (qy * qy + qz * qz));
+
+	return (euler_x, euler_y, euler_z);
+}
+
+/// Convert from AxisAngle to Euler YXZ.
+///
+/// All rotations are expected and returned as radians.
+///
+/// # Arguments
+///
+/// * `rx` - The X component of the axis angle vector.
+/// * `ry` - The Y component of the axis angle vector.
+/// * `rz` - The Z component of the axis angle vector.
+/// * `angle` - The rotation around the axis angle vector to apply in radians.
+///
+/// Known robots that use this format:
+fn convert_axis_angle_to_euler_yxz(rx: f64, ry: f64, rz: f64, angle: f64) -> (f64, f64, f64) {
+	let axis_norm = sqrt(rx * rx + ry * ry + rz * rz);
+
+	// Compute the quaternion
+	let sin_angle = sin(0.5 * angle);
+	let qw = cos(0.5 * angle);
+	let qx = rx / axis_norm * sin_angle;
+	let qy = ry / axis_norm * sin_angle;
+	let qz = rz / axis_norm * sin_angle;
+
+	let euler_x = asin(2.0 * (qw * qx - qy * qz));
+	let euler_y = atan2(2.0 * (qx * qz + qw * qy), 1.0 - 2.0 * (qx * qx + qy * qy));
+	let euler_z = atan2(2.0 * (qx * qy + qw * qz), 1.0 - 2.0 * (qx * qx + qz * qz));
+
+	return (euler_x, euler_y, euler_z);
+}
+
+/// Convert from AxisAngle to Euler ZXY.
+///
+/// All rotations are expected and returned as radians.
+///
+/// # Arguments
+///
+/// * `rx` - The X component of the axis angle vector.
+/// * `ry` - The Y component of the axis angle vector.
+/// * `rz` - The Z component of the axis angle vector.
+/// * `angle` - The rotation around the axis angle vector to apply in radians.
+///
+/// Known robots that use this format:
+fn convert_axis_angle_to_euler_zxy(rx: f64, ry: f64, rz: f64, angle: f64) -> (f64, f64, f64) {
+	let axis_norm = sqrt(rx * rx + ry * ry + rz * rz);
+
+	// Compute the quaternion
+	let sin_angle = sin(0.5 * angle);
+	let qw = cos(0.5 * angle);
+	let qx = rx / axis_norm * sin_angle;
+	let qy = ry / axis_norm * sin_angle;
+	let qz = rz / axis_norm * sin_angle;
+
+	let euler_x = asin(2.0 * (qy * qz + qw * qx));
+	let euler_y = atan2(2.0 * (qw * qy - qx * qz), 1.0 - 2.0 * (qx * qx + qy * qy));
+	let euler_z = atan2(2.0 * (qw * qz - qx * qy), 1.0 - 2.0 * (qx * qx + qz * qz));
+
+	return (euler_x, euler_y, euler_z);
+}
+
+/// Convert from AxisAngle to Euler ZYX.
+///
+/// All rotations are expected and returned as radians.
+///
+/// # Arguments
+///
+/// * `rx` - The X component of the axis angle vector.
+/// * `ry` - The Y component of the axis angle vector.
+/// * `rz` - The Z component of the axis angle vector.
+/// * `angle` - The rotation around the axis angle vector to apply in radians.
+///
+/// Known robots that use this format:
+/// - Mitsubishi Industrial Robot CR750/CR751 Series
```

Comment:
```suggestion
/// Known robots that use this format:
/// - Mitsubishi Industrial Robot CR750/CR751 Series
/// - Kawasaki, after conversion to degrees
/// - Kuka, after conversion to degrees
```
Source: https://doc.rc-visard.com/v21.07/en/pose_format_kuka.html#:~:text=KUKA%20robots%20use%20the%20so,)rx(C).

Reply:
Reworded it a bit, but done :+1: